src/main/java/ch/njol/skript/ScriptLoader.java

```diff
@@ -439,98 +414,62 @@ private static <T> CompletableFuture<T> makeFuture(Supplier<T> supplier, OpenClo
-				if (isAsync()) {
-					oldLoadedFiles.removeAll(loadedFiles);
-					for (File script : oldLoadedFiles) {
-						if (script == null)
-							throw new NullPointerException();
-						
-						// Use internal unload method which does not call validateFunctions()
-						unloadScript_(script);
-						String name = Skript.getInstance().getDataFolder().toPath().toAbsolutePath()
-							.resolve(Skript.SCRIPTSFOLDER).relativize(script.toPath()).toString();
-						assert name != null;
-						Functions.clearFunctions(name);
-					}
-					Functions.validateFunctions(); // Manually validate functions
-				}
-				
-				if (scriptInfo.files == 0)
-					Skript.warning(m_no_scripts.toString());
-				if (Skript.logNormal() && scriptInfo.files > 0)
-					Skript.info(m_scripts_loaded.toString(
-						scriptInfo.files,
-						scriptInfo.triggers,
-						scriptInfo.commands,
-						start.difference(new Date())
-					));
-			});
+	 * Loads the Script present at the file using {@link #loadScripts(List, OpenCloseable)},
+	 * 	sending info/error messages when done.
+	 * @param file The file to load. If this is a directory, all scripts within the directory and any subdirectories will be loaded.
+	 * @param openCloseable An {@link OpenCloseable} that will be called before and after
+	 *                         each individual script load (see {@link #makeFuture(Supplier, OpenCloseable)}).
+	 */
+	public static CompletableFuture<ScriptInfo> loadScripts(File file, OpenCloseable openCloseable) {
+		return loadScripts(loadStructures(file), openCloseable);
+	}
+
+	/**
+	 * Loads the Scripts present at the files using {@link #loadScripts(List, OpenCloseable)},
+	 * 	sending info/error messages when done.
+	 * @param files The files to load. If any file is a directory, all scripts within the directory and any subdirectories will be loaded.
+	 * @param openCloseable An {@link OpenCloseable} that will be called before and after
+	 *                         each individual script load (see {@link #makeFuture(Supplier, OpenCloseable)}).
+	 */
+	public static CompletableFuture<ScriptInfo> loadScripts(Collection<File> files, OpenCloseable openCloseable) {
+		return loadScripts(files.stream()
+			.sorted()
+			.map(ScriptLoader::loadStructures)
+			.flatMap(List::stream)
+			.collect(Collectors.toList()), openCloseable);
 	}
 	
 	/**
 	 * Loads the specified scripts.
 	 *
-	 * @param configs Configs for scripts, loaded by {@link #loadStructures(File[])}
+	 * @param configs Configs representing scripts.
 	 * @param openCloseable An {@link OpenCloseable} that will be called before and after
 	 *                         each individual script load (see {@link #makeFuture(Supplier, OpenCloseable)}).
 	 * @return Info on the loaded scripts.
 	 */
 	public static CompletableFuture<ScriptInfo> loadScripts(List<Config> configs, OpenCloseable openCloseable) {
-		AtomicBoolean syncCommands = new AtomicBoolean();
-    
+		if (configs.isEmpty()) // Nothing to load
```

Comment:
Any errors will have been printed, but maybe it could be a good idea for locations like 
https://github.com/SkriptLang/Skript/blob/4379be86a4e594d1c7c7ee488da06991b53c820e/src/main/java/ch/njol/skript/SkriptCommand.java#L140-L141

Thoughts? @TPGamesNL

Reply:
I think it's fine to have it valid behaviour to load 0 scripts. However, I think it's also a good idea to add a warning when it's used-initiated (through return value ScriptInfo#files). We currently have one in Skript (`Skript.m_no_scripts` usages), but it is only used during Skript enabling.